modules/structs/commit_status.go

```diff
@@ -24,18 +24,15 @@ const (
 
 // NoBetterThan returns true if this State is no better than the given State
 func (css CommitStatusState) NoBetterThan(css2 CommitStatusState) bool {
-	switch css {
-	case CommitStatusError:
-		return true
-	case CommitStatusFailure:
-		return css2 != CommitStatusError
-	case CommitStatusWarning:
-		return css2 != CommitStatusError && css2 != CommitStatusFailure
-	case CommitStatusPending:
-		return css2 != CommitStatusError && css2 != CommitStatusFailure && css2 != CommitStatusWarning
-	default:
-		return css2 != CommitStatusError && css2 != CommitStatusFailure && css2 != CommitStatusWarning && css2 != CommitStatusPending
+	commitStatusPriorities := map[CommitStatusState]int{
```

Comment:
Maybe this can be defined on the outside of this function, then it can be used by other functions in the future.

Reply:
I don't think so, the comparison of priorities between statuses only relies on the `NoBetterThan` function, not `commitStatusPriorities`.